src/QtLocationPlugin/MapProvider.h

```diff
@@ -40,16 +40,19 @@ class QNetworkRequest;
 
 class MapProvider : public QObject
 {
-    Q_OBJECT
-
 public:
-    MapProvider(const QString& referrer, const QString& imageFormat, quint32 averageSize = AVERAGE_TILE_SIZE,
-                QGeoMapType::MapStyle mapStyle = QGeoMapType::CustomMap, QObject* parent = nullptr);
+    MapProvider(const QString &mapName, const QString &referrer, const QString &imageFormat, quint32 averageSize = AVERAGE_TILE_SIZE,
+                QGeoMapType::MapStyle mapStyle = QGeoMapType::CustomMap);
+    virtual ~MapProvider();
 
     QString getImageFormat(QByteArrayView image) const;
 
+    // TODO: Download Random Tile And Use That Size Instead?
     quint32 getAverageSize() const { return _averageSize; }
+
     QGeoMapType::MapStyle getMapStyle() const { return _mapStyle; }
+    QString getMapName() const { return _mapName; }
```

Comment:
```suggestion
    const QString& getMapName() const { return _mapName; }
```

Reply:
Isn't this no longer necessary as of C++17?

Reply:
My understanding is that copy elision would not be applied here as it's not returning a temporary.
And inlining would optimize some cases, but not generally all. E.g. `QString mapName = x.getMapName();`. You could argue that you should use `const QString& mapName = x.getMapName();` in that case too, where I think the compiler would remove all copies, but I'm not sure if it's guaranteed. Maybe you're more familiar with the details here?